### PR TITLE
ROX-18978, ROX-20225: Policy updates

### PR DESCRIPTION
**Pull request by theencee**
## Description

Combining two changes into one PR because they all deal with one migration.

The changes include:

* Update the policy migrator code to make it a little more efficient (avoid an unnecessary Exists call). Unfortunately this does lead to some code duplication, but given that the code is used in old migrations, it's best to be careful.
* Some policies have duplicate exclusions for some reason which used to cause issues with migrator. That's fixed now.

And the actual policy changes:

- ROX-18978: Updates two policies that check for `iptables` process to also check `nft` (which is for nftables)
- ROX-20225: Exclude deployment `aide-worker-fileintegrity` in namespace `openshift-file-integrity` from the policy "Process with UID 0"

If you are only interested in reviewing the policy, I recommend looking at only the files in the pkg/default/policies/files directory since that's the actual policies. Everything else is scaffolding for migrations.

## Checklist
- [ ] Investigated and inspected CI test results
- [x] Unit test and regression tests added
~[ ] Evaluated and added CHANGELOG entry if required~
~[ ] Determined and documented upgrade steps~
~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

Manual testing

1. Installed fresh version of ACS and validated all the changes
2. Start with 4.2.2 and then upgraded to this build. Validated all the changes
3. Start with 4.2.12 Upgrade to this build. Rollback to 4.2.2. Roll forward to this version.

For #3 the most important thing is that central didn't error out or crash. On revert, all of the changes (new criteria, and exclusion) all remained. But that should be ok. On roll forward, the policy criteria remain. The exclusion unfortunately generates a duplicate, but I think that's a side effect we can take (especially since some existing policies already have dupes for exclusions). 

Screenshot 2023-10-26 at 5 01 25 PM

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: migrator/migrations/m_187_to_m_188_add_nftables_to_policy/migration.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		},
+	}
+)
+
+func updatePolicies(db *gorm.DB) error {
+
+	return policymigrationhelper.MigratePoliciesWithDiffsAndStore(
+		policyDiffFS,
+		policyDiffs,
+		// Check if policy with specified id exists
```
Comment (dashrews78) [inline migrator/migrations/m_187_to_m_188_add_nftables_to_policy/migration.go:57]: This is virtually the same as the func below that does the get.  So I'm wondering if we even need the Count func as we can probably satisfy its purpose with the Get func (depending on how you wire the rest of it up).  Aside from that the usage of gorm appears to be correct to me.

File: migrator/migrations/m_192_to_m_193_policy_updates_for_4_3/migration.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	"github.com/stackrox/rox/migrator/migrations"
+	"github.com/stackrox/rox/migrator/migrations/m_192_to_m_193_policy_updates_for_4_3/schema"
+	"github.com/stackrox/rox/migrator/migrations/policymigrationhelper"
+	"github.com/stackrox/rox/migrator/types"
+	"gorm.io/gorm"
+)
+
+var (
+	migration = types.Migration{
+		StartingSeqNum: 192,
```
Comment (dashrews78) [inline migrator/migrations/m_192_to_m_193_policy_updates_for_4_3/migration.go:18]: For your awareness there are at least 3 other PRs out right now using this migration number.  So we will need to keep our eyes open for when those land and adjust if necessary.

**Review by dashrews78 — APPROVED** [review]
Code changes look good.  Please have someone else review the policies.  
Before merging make sure no one else took migration 192 as @ksurabhi91 and @rhybrillou have PRs out using 192.

File: migrator/migrations/policymigrationhelper/policy_migrator.go [hunk 3]
```diff
@@ -101,6 +105,20 @@ func (u *PolicyUpdates) applyToPolicy(policy *storage.Policy) {
...
+	if u.CategoriesToRemove != nil {
+		for _, toRemove := range u.CategoriesToRemove {
+			if !removeCategory(policy, toRemove) {
+				log.WriteToStderrf("policy ID %s has already been altered because category was already removed. Will not update.", policy.Id)
+				continue
+			}
+		}
+	}
+
+	// Add new exclusions as needed
```
Comment (clickboo) [inline migrator/migrations/policymigrationhelper/policy_migrator.go]: nit: comment: Remove categories as needed

**Review by srcporter — APPROVED** [review]
looks great!

**Comment by theencee** [comment]
Removed all changes to do with categories for this PR

**Comment by theencee** [comment]
The ocp-4-10-qa-e2e-tests failure looks like it's due to a flaky AuditLogsAlertsTest which we know about and is unrelated.

**Pull request merged**